### PR TITLE
util/plancodec: migrate test-infra to testify

### LINKED ISSUE #26518: migrate test-infra to testify for `util/plancodec` pkg

### PR DESCRIPTION
**Pull request by beyoung**
### What problem does this PR solve?

Issue Number: close #26518

### Release note

```release-note
None
```

### CODE REVIEW

File: util/plancodec/codec_test.go [hunk 1]
```diff
@@ -14,45 +14,45 @@
 package plancodec
 
 import (
-	. "github.com/pingcap/check"
 	"github.com/pingcap/tidb/kv"
+	"github.com/stretchr/testify/require"
+	"testing"
 )
```
Comment (tisonkun) [inline util/plancodec/codec_test.go:20]: These imports violate goimports rule, you may setup an auto importer in GoLand (if you use it) as

image

File: util/plancodec/codec_test.go [hunk 2]
```diff
@@ -14,45 +14,45 @@
...
+	t.Parallel()
 	cases := []encodeTaskTypeCase{
 		{true, kv.UnSpecified, "0", "root"},
 		{false, kv.TiKV, "1_0", "cop[tikv]"},
 		{false, kv.TiFlash, "1_1", "cop[tiflash]"},
 		{false, kv.TiDB, "1_2", "cop[tidb]"},
 	}
 	for _, cas := range cases {
-		c.Assert(EncodeTaskType(cas.IsRoot, cas.StoreType), Equals, cas.EncodedStr)
+		require.Equal(t, EncodeTaskType(cas.IsRoot, cas.StoreType), cas.EncodedStr)
```
Comment (tisonkun) [inline util/plancodec/codec_test.go:38]:
```suggestion
		require.Equal(t, cas.EncodedStr, EncodeTaskType(cas.IsRoot, cas.StoreType))
```

Please be aware different order of position for `expected` and `actual` args.

Comment (tisonkun) [reply util/plancodec/codec_test.go:38]: ditto others.

**Review by tisonkun — CHANGES_REQUESTED** [review]
Thanks for your contribution @beyoung ! Generally looks good, only you may take care of import order and args position. Comments inline.

File: util/plancodec/codec_test.go (hunk 2 above)
Comment (beyoung) [reply util/plancodec/codec_test.go]: @tisonkun Thx, DONE!

File: util/plancodec/id_test.go [hunk 3]
```diff
@@ -16,67 +16,71 @@ package plancodec
...
-	c.Assert(typeTableSampleID, Equals, 42)
-	c.Assert(typeTableFullScan, Equals, 43)
-	c.Assert(typeTableRangeScan, Equals, 44)
-	c.Assert(typeTableRowIDScan, Equals, 45)
-	c.Assert(typeIndexFullScan, Equals, 46)
-	c.Assert(typeIndexRangeScan, Equals, 47)
-	c.Assert(typeExchangeReceiver, Equals, 48)
-	c.Assert(typeExchangeSender, Equals, 49)
+	t.Parallel()
+	testCases := []struct {
```
Comment (tisonkun) [inline util/plancodec/id_test.go:25]: clean works!

**Review by tisonkun — APPROVED** [review]
LGTM.

File: util/plancodec/id_test.go [hunk 4]
```diff
@@ -16,67 +16,71 @@ package plancodec
...
-	c.Assert(typeLoadDataID, Equals, 41)
-	c.Assert(typeTableSampleID, Equals, 42)
-	c.Assert(typeTableFullScan, Equals, 43)
-	c.Assert(typeTableRangeScan, Equals, 44)
-	c.Assert(typeTableRowIDScan, Equals, 45)
-	c.Assert(typeIndexFullScan, Equals, 46)
-	c.Assert(typeIndexRangeScan, Equals, 47)
-	c.Assert(typeExchangeReceiver, Equals, 48)
-	c.Assert(typeExchangeSender, Equals, 49)
+	t.Parallel()
```
Comment (beyoung) [inline util/plancodec/id_test.go:24]: Should I move `t.Parallel()` to loop below ?
image
@tisonkun

File: util/plancodec/id_test.go (hunk 3 above)
Comment (beyoung) [reply util/plancodec/id_test.go:25]: anything need I do to merge this pr? @tisonkun

**Comment by tisonkun** [comment]
@beyoung thanks for ping me. I will cc reviewers for this PR.

/cc @xhebox @tiancaiamao @mmyj

**Review by xhebox — APPROVED** [review]
Not a reviewer of planner.

**Review by winoros — APPROVED** [review]

**Review by francis0407 — APPROVED** [review]
LGTM

**Pull request merged**